core/src/main/java/hudson/model/TaskListener.java

```diff
@@ -111,4 +112,6 @@
      * {@link TaskListener} that discards the output.
      */
     TaskListener NULL = new StreamTaskListener(new NullStream());
+
+    Charset getCharset();
```

Comment:
Was the reason I've been asking we migrate jenkins 2.0 to JDK 8 ...
I'm open to suggestion on way to introduce this without binary incompatibility